include/multipass/id_mappings.h

```diff
@@ -18,12 +18,40 @@
 #ifndef MULTIPASS_ID_MAPPINGS_H
 #define MULTIPASS_ID_MAPPINGS_H
 
+#include <multipass/format.h>
+#include <multipass/logging/log.h>
+
+#include <algorithm>
+#include <iterator>
+#include <set>
 #include <utility>
 #include <vector>
 
+namespace mpl = multipass::logging;
+
 namespace multipass
 {
 typedef typename std::vector<std::pair<int, int>> id_mappings;
+
+template <class IDMap>
+id_mappings unique_id_mappings(const IDMap& xid_mappings)
+{
+    id_mappings ret;
+    std::set<std::pair<int, int>> id_set;
+
+    auto pred = [&id_set](const auto& m) {
```

Comment:
Would you mind renaming to `is_mapping_repeated`?

Reply:
Done.